Velaptor/OpenGL/GLWindow.cs

```diff
@@ -369,21 +465,74 @@ private void GameWindow_RenderFrame(object? sender, FrameTimeEventArgs e)
+        /// <param name="arg3">Additional argument from OpenGL.</param>
+        private void GLKeyboardInput_KeyDown(IKeyboard keyboard, Key key, int arg3) => this.keyboardInput.SetState((KeyCode)key, true);
+
+        /// <summary>
+        /// Invoked when any keyboardInput key transitions from the down position to the up position.
         /// </summary>
-        private void RegisterEvents()
+        /// <param name="keyboard">The system keyboardInput.</param>
+        /// <param name="key">The key that was released.</param>
+        /// <param name="arg3">Additional argument from OpenGL.</param>
+        private void GLKeyboardInput_KeyUp(IKeyboard keyboard, Key key, int arg3) => this.keyboardInput.SetState((KeyCode)key, false);
+
+        /// <summary>
+        /// Invoked when any of the mouse buttons are pressed into the down position over the window.
+        /// </summary>
+        /// <param name="mouse">The system mouse object.</param>
+        /// <param name="button">The button that was pushed down.</param>
+        private void GLMouseInput_MouseDown(IMouse mouse, SilkMouseButton button) => this.mouseInput.SetState((VelaptorMouseButton)button, true);
+
+        /// <summary>
+        /// Invoked when any of the mouse buttons are released from the down position into the up position over the window.
+        /// </summary>
+        /// <param name="mouse">The system mouse object.</param>
+        /// <param name="button">The button that was pushed down.</param>
+        private void GLMouseInput_MouseUp(IMouse mouse, SilkMouseButton button) => this.mouseInput.SetState((VelaptorMouseButton)button, false);
+
+        /// <summary>
+        /// Invoked when there is mouse scroll wheel input.
+        /// </summary>
+        /// <param name="mouse">The system mouse object.</param>
+        /// <param name="wheelData">Positional data about the mouse scroll wheel.</param>
+        private void GLMouseInput_MouseScroll(IMouse mouse, ScrollWheel wheelData)
         {
-            this.windowFacade.Load += GameWindow_Load;
-            this.windowFacade.Unload += GameWindow_Unload;
-            this.windowFacade.UpdateFrame += GameWindow_UpdateFrame;
-            this.windowFacade.RenderFrame += GameWindow_RenderFrame;
-            this.windowFacade.Resize += GameWindow_Resize;
+            this.mouseInput.SetScrollWheelSpeed((int)wheelData.Y);
+
+            var wheelDirection = wheelData.Y switch
+            {
+                > 0 => MouseScrollDirection.ScrollUp,
+                < 0 => MouseScrollDirection.ScrollDown,
+                _ => MouseScrollDirection.None
+            };
+
+            this.mouseInput.SetScrollWheelDirection(wheelDirection);
+        }
+
+        /// <summary>
+        /// Invoked when the mouse moves over the window.
+        /// </summary>
+        /// <param name="mouse">The system mouse object.</param>
+        /// <param name="position">The position of the mouseInput.</param>
```

Comment:
Refactor comment to below

```suggestion
        /// <param name="position">The position of the mouse.</param>
```